visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -48,6 +51,29 @@ export class NodeDecorator {
 			})
 	}
 
+	public static doesExclusionResultInEmptyMap(map: CodeMapNode, blacklist: BlacklistItem[]): boolean {
+		let excludedNodes = 0
+		const flattened = ignore()
+		const excluded = ignore()
+
+		for (const item of blacklist) {
+			const path = CodeMapHelper.transformPath(item.path)
+			item.type === BlacklistType.flatten ? flattened.add(path) : excluded.add(path)
+		}
+
+		const tree = hierarchy(map)
+		const desc = tree.descendants()
+
+		desc.map(node => {
```

Comment:
If `desc` is an array, it's better to use `some` or to use a plain for loop. If it's an iterator, using a for loop would also be best, since it's possible to stop the loop as soon as the excludedNodes exceed the tree leaves length.

Reply:
I agree, that a for-loop is benificial since you can exit it earlier, but in this case, the number of excluded nodes can never exceed the number of nodes in the map. It's either all nodes are excluded or less.
I think that a for-loop is easier to read though and changed it. I also refactored, that we count the leaves as we traverse the tree. This should be more efficient than calling `hierarchy(map).leaves()`, since `leaves()` traverses the whole tree again.

```js
export default function() {
  var leaves = [];
  this.eachBefore(function(node) {
    if (!node.children) {
      leaves.push(node);
    }
  });
  return leaves;
}

```